### ISSUE TITLE
A11y - Modal is having tabIndex of -1 and this prevents any focus when using keyboard

### ISSUE THREAD

**Issue by ashpc** [IBM/carbon-components-react#594]
## Detailed description

Describe in detail the issue you're having. Is this a feature request (new component, new icon), a bug, or a general issue?

When using keyboard to open a modal the focus is not transferred to the modal as the tabIndex is set to -1 instead of being a prop.

> Is this issue related to a specific component?

Modal

> What did you expect to happen? What happened instead? What would you like to see changed?

When using keyboard to open a modal the focus should be on the modal and not to the elements below the modal. When a prop is passed then use it for the tabIndex if not set to a default value of -1.

> What browser are you working in?

All browsers.

> What version of the Carbon Design System are you using?

 "carbon-components": "8.10.0",
 "carbon-components-react": "5.17.2"

## Steps to reproduce the issue

1. Use keyboard to open a modal
2. Once modal is displayed then hit tab, you will see that the focus is moved to the next element below the modal and not the modal elements.

## Add labels
Please choose the appropriate label(s) from our existing label list to ensure that your issue is properly categorized. This will help us to better understand and address your issue.
bug

**Comment by marijohannessen** [comment]
I'm having problems replicating this bug, did you test it directly in our Storybook or do you have an example of your code somewhere so I can take a look @ashpc ?

**Comment by tw15egan** [comment]
Closing due to inactivity / unable to reproduce. When I tab after opening a modal via keyboard, the next focus is on the `x` icon.

**Comment by ashpc** [comment]
@tw15egan and @marijohannessen here are the Steps to reproduce issue:
1)  Go to IAM UI at https://console.bluemix.net/iam/#/serviceids
2) Click on Create button using keyboard/mouse,
3) Notice focus is on Create button
4) Now when you hit tab
**Problem:** Focus is on the sorting arrow of the table
**Expected Behaviour**:  Focus moves to modal component 

Let me know if you have questions. It would be great if this is re-opened.

**Comment by tw15egan** [comment]
I'm still unable to reproduce, the focus goes to the `x` once I open the modal with the keyboard.

**Comment by marijohannessen** [comment]
I am still unable to reproduce as well with the given steps.

**Comment by marijohannessen** [comment]
With a table created, I am able to reproduce the issue.

**Comment by marijohannessen** [comment]
@ashpc After investigating this further, I can see now that the button that triggers the modal is in the same container as the table and not a sibling of the modal wrapper. I think this is what's causing this issue cause I am unable to reproduce this looking at our examples. It fixes the problem if the modal wrapper is placed after the modal trigger button and not after the table container. So in the DOM it needs to look like this:

```
<div class="table-right-actions">
   <button tabindex="0" class="bx--btn bx--btn--sm bx--btn--primary" type="button">...</button>
   <div class="modal-wrapper" role="dialog" aria-label>...</div>
</div>
```

It will require some re-styling, but it seems like removing `display: inline-block` and `vertical-align: middle` from the `.modal-wrapper` fixes most of it.

**Comment by tw15egan** [comment]
Error fixed on user end
